modules/videoio/src/cap_msmf.cpp

```diff
@@ -1204,8 +1627,147 @@ bool CvCapture_MSMF::retrieveFrame(int, cv::OutputArray frame)
     return false;
 }
 
+bool CvCapture_MSMF::retrieveAudioFrame(int index, cv::OutputArray frame)
+{
+    CV_TRACE_FUNCTION();
+    do
+    {
+        if (audioSamples.empty())
+            break;
+        else
+        {
+            for (auto item : audioSamples)
+            {
+                if (!item)
+                    break;
+            }
+        }
+        _ComPtr<IMFMediaBuffer> buf = NULL;
+        BYTE* useAudioData = NULL;
+        BYTE* ptr = NULL;
+        DWORD maxsize = 0, cursize = 0;
+        CV_TRACE_REGION("get_contiguous_buffer");
+        for (auto item : audioSamples)
+        {
+            if (!SUCCEEDED(item->ConvertToContiguousBuffer(&buf)))
+            {
+                CV_TRACE_REGION("get_buffer");
+                DWORD bcnt = 0;
+                if (!SUCCEEDED(item->GetBufferCount(&bcnt)))
+                    break;
+                if (bcnt == 0)
+                    break;
+                if (!SUCCEEDED(item->GetBufferByIndex(0, &buf)))
+                    break;
+            }
+
+            if (!SUCCEEDED(buf->Lock(&ptr, &maxsize, &cursize)))
+                break;
+            for (unsigned int i = 0; i < cursize; i++)
+            {
+                bufferAudioData.push_back(*(ptr+i));
+            }
+            CV_TRACE_REGION_NEXT("unlock");
+            buf->Unlock();
+            buf = NULL;
+        }
+        audioSamples.clear();
+        int numberOfByte = (videoStream != -1) ? (int)((double)(curVideoTime/1e7)*captureAudioFormat.nSamplesPerSec*captureAudioFormat.nChannels*(captureAudioFormat.bit_per_sample)/8) : cursize;
```

Comment:
> curVideoTime/1e7

It is not accurate with integer numbers:

```
LONGLONG curVideoTime;
```

Division should be the last operation of expression (also avoid overflow - `int64_t` should be enough).

Reply:
> > curVideoTime/1e7
> 
> It is not accurate with integer numbers:
> 
> ```
> LONGLONG curVideoTime;
> ```
> 
> Division should be the last operation of expression (also avoid overflow - `int64_t` should be enough).

Mat accepts arguments with the int type. If we take int64_t numberOfByte, then the conversion of int64_t types to int will occur in the Mat constructor.

Reply:
- compute as int64_t
- avoid doubles / floating-point computations
- check range of final result and cast to "int" if needed